README
Fldigi is a software modem for Amateur Radio use. It is a sound card based
program that is used for both transmitting and receiving data in any of the
following modes:

BPSK and QPSK        31, 63, 125, 250 (both), and 63F and 500 (BPSK only)
PSKR                 125, 250, and 500
CW                   speeds from 5 to 200 wpm
DominoEX             4, 5, 8, 11, 16 and 22; also with FEC
FSQ
Hellschreiber        Feld Hell, Slow Hell, Hell x5/x9, FSKHell(-105) and Hell 80
IFKP
MFSK                 4, 8, 11, 16, 22, 31, 32 and 64; most with image support
MT63                 500, 1000 and 2000
OLIVIA               various tones and bandwidths
RTTY                 various baud rates, shifts, nbr. of data bits, etc.
THOR                 4, 5, 8, 11, 16 and 22
Throb and ThrobX     1, 2, and 4
WWV                  receive only - calibrate your sound card to WWV
Frequency Analysis   receive only - measure the frequency of a carrier

Fldigi can also control a transceiver using Hamlib or RigCAT I/O, perform online
or cdrom QRZ queries, log QSOs with the built-in logbook or Xlog, and send
reception reports to the PSK Automatic Propagation Reporter.

The latest version can always be found at:

  http://www.w1hkj.com/

Visit this page for extensive documentation and an archive of XML files for
transceivers supported by RigCAT.

The wiki page contains FAQs and HOWTOs, as well as links to all Fldigi
resources:

  https://fedorahosted.org/fldigi/

The GIT repository can be found at:

  https://sourceforge.net/p/fldigi/fldigi/ci/master/tree/

For support, news and updates, join one or more of the following mailing lists
and Yahoo groups.  Stable releases are announced to all lists and groups; test
(alpha) releases are announced to fldigi-alpha and fldigi-announce.

  * linuxham group
    General discussion on fldigi, related software and other ham radio topics
    https://groups.io/g/linuxham

  * win-fldigi group
    For Windows users of fldigi
    http://groups.yahoo.com/group/win-fldigi/

  * NBEMSham group
    Special focus on NBEMS operation
    https://groups.io/g/nbems

  * fldigi-alpha list
    Discussion of fldigi testing and related subjects
    https://sourceforge.net/p/fldigi/mailman/fldigi-alpha/

  * fldigi-devel list
    Fldigi development topics
    https://sourceforge.net/p/fldigi/mailman/fldigi-devel/


NEWS
NEWS for version 4.x

Version 4.0.4

  cw scanner
    . adds multi signal scanner / browser
  Sound Card
    . keep input and output sound card devices open for duration of
      fldigi execution
    . Corrected tune output device
    . Changed tune, wfid, video and cw id to use tx offset
  CW QSY
    . Correct QSY when transceiver in CW and CW-R type modes
  CALL field size
    . Increased CALL, OP_CALL, STA_CALL from 15 to 30 chars
  LOG area controls
    . Resize controls to add County entry control.  Allow user
      to select which extra qso field is displayed; country or
      county.

Version 4.0.3

  CW code update
    . Changed all element timing to number of samples
      both CW and CW scanning detectors
    . Removed FIR filter. FFT filter only for CW signal path.
    . Removed Hilbert transform from CW signal path
    . Update CW documentation

  OS X Wizard
    . Discovered cause of Wizard failure on OS X
      - newly created dialogs cannot be accessed until actually shown.

  CW wait
    . Correct Windows performance at end of CW Tx text
      - program should wait for keyboard input
      - terminate with the ^r character pair
      - terminate with control-r
      - terminate with the T/R button

Version 4.0.2

  Revert "wefax"
    . This reverts commit 5609415222335e9376defc325f20ae29c5c8c8ca.
    . Modifies wefax.cxx iaw request from Remi, wefax author

  Lotw Match
    . fix failure to mark logbook as dirty when records are matched

  flrigi bw synch
    . Improve bandwidth table / selection sync with flrig when changing 
      xcvr mode from fldigi

  Clang 8.1.0
    . Correct warning notification with latest Clang compiler on
      macOS Sierra 10.12.4

  Sound card
    . Changed to separate Rx/Tx soundcard instances
      - only open/close when samplerate changes
      - attempt to reduce incidence of lost sound card in OS X

  IFKP image processing
    . Rx processing failed to return to TEXT state upon conclusion of 
      image transfer

  Macro Save
    . Corrected issue with macro tag "Save"

  CW end of transmit
    . Add <LF> character to Rx echo text when going from Tx to Rx
      - provides visual separation of Tx and Rx text in the Rx panel
      - color rendition still applies.

  Font Initialization
    . Force all font initialization to be name dependent
    . Font name / Font number can change as user installs or removes 
      fonts from the OS with potentially fatal impact on fldigi font usage.
    . May require user to reconfigure various font selections

  OS X socket connect
    . connect test for fail conditions not correct for OS X

  Winkeyer exit process
    . Release resources when closing Winkeyer
    . Issue ADMIN:CLOSE command to Winkeyer when disconnecting or
      closing fldigi.

Version 4.0.1

  Socket connect
    . FreeBSD mod causes seg fault on non BSD platforms
      - make change OS dependent

Version 4.0.0

  OS X 10.12 build
    . Compile changes for OS X Darwin 10.12 Clang build

  STATION_CALLSIGN
    . Change adif field STATION_CALL to STATION_CALLSIGN compliant with
      adif specification

  Free BSD mods
    . Compliance mods for current version of Clang compiler

  Logbook mods
    . Update log field LOTWSDATE for each record sent to LOTW when the
      upload file is created
    . clear LOTWSDATE field if lotw rejects submission or fails to 
      respond within 25 seconds
    . insure that lotw review dialog closes on program exit
    . restore tabular sort order when adding a record

  Default Tx Audio
    . Change default audio output to both channels

  remove sprintf
    . change all remaining occurances of sprintf to snprintf

  Documentation update
    . New documentation for winkeyer interface

  n3fjp logger search
    . set initial search for n3fjp server to 5 second intervals

  fopen->fl_fopen
    . update to UDF-8 compatible fopen call

  48 bar macro
    . allow user to select above / below Rx/Tx panels

  CW decoder
    . Improved CW decoder
      - removed user adjustable signal level controls
      - added automatic tracking filters for Rx signal level
      - tracking filters change parameters with detected WPM.

  DX cluster
    . Update dxcluster QSY to correctly align signal to sweet spot
      when transceiver is in a CW mode.
    . add ability to respond to password prompt

  Logbook support
    . Add checkband test to new log entry before writing record
    . Added BAND control to log view, contest tab
      - fldigi does not use the BAND field, it is present as a
        convenience for exporting or importing from log programs
        that do use it.
      - the control is on the contest tab since that seems to be
        the most common use for BAND (e.g. field day event).
    . OS X file naming dialog drops the ".adif" extension.
      - add code to insure newlog etc have the correct extension.

  WinKeyer
    . Added embedded WinKeyer interface

  fsq ifkp s/n computation
    . Additional change to s/n measurement

  fltk version update
    . update all fluid generated source and header files

  Macro Date Time Formats
    . Allow user to specify the date time format using strftime format
      strings
      - <ZT> becomes <ZT:"%H%M%S">; <ZT:"%H:%M:%SZ"> etc.
      - <ZT:fmt> fmt is the strftime format string
        if the ':fmt' is not in the macro tag then the current format
        is used

  lotw addition
    . Add jt9 modes for lotw upload
    . Allow up to 7 numerals in frequency element

  THOR/IFKP Image transfers
    . Add auto save Rx images to
      - Thor
      - IFKP
    . Corrected IFKP send image sequencing
      - pic%[T] where T is image scale was not being sent before the
        image fascimille.
    . Added Thor and IFKP image transfer using the <IMAGE:fname> macro
      tag

  ifkp s/n update
    . Change to s/n measurement
      - noise measured over full range of possible signal bins
      - noise referenced to 3000 Hz bandwidth

  fsq s/n update

src/config.h
// ----------------------------------------------------------------------------
// config.h  --  user configuration items for fldigi
//
// Copyright (C) 2006
//		Dave Freese, W1HKJ
//
// This file is part of fldigi.  Adapted in part from code contained in gmfsk 
// source code distribution.
//
// fldigi is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 2 of the License, or
// (at your option) any later version.
//
// fldigi is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with fldigi; if not, write to the Free Software
// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
// ----------------------------------------------------------------------------

#ifndef _CONFIG_H
#define _CONFIG_H

// You can change the x1 width of the waterfall / spectrum display by modifying this
// constant.
// Suggest that you make the value a multiple of 100.
// DO NOT EXCEED 4000
// The larger the number the greater the cpu load will be for creating the
// waterfall display
#define DEFAULT_IMAGE_WIDTH 3000
#define DEFAULT_HWFALL 140
#define DEFAULT_HNOM 570
#define DEFAULT_WNOM (DEFAULT_IMAGE_WIDTH / 4 + DEFAULT_HWFALL - BTN_HEIGHT)

#define EMC_HWFALL 120
#define EMC_HNOM 400
#define EMC_WNOM (EMC_HWFALL + 580)

extern int IMAGE_WIDTH;
extern int Hwfall;
extern int HNOM;
extern int WNOM;

// widget sizes internal to the waterfall widget
#define BEZEL		 2
#define WFTEXT		10
#define WFSCALE     10
#define WFMARKER     6
#define BTN_HEIGHT	20

#define bwColor		30
#define bwFFT		30
#define bwX1		30
#define bwMov		20
#define cwCnt		100
#define cwRef		50
#define cwMode		85
#define bwQsy		45
#define bwRate		45
#define bwXmtLock	45
#define bwRev		45
#define bwXmtRcv	45
#define wSpace		2

#define Hmenu		22
#define Hqsoframe	48
#define Hnotes		22
#define Hstatus		22
#define Hmacros		22

#define Htext		(HNOM - 4 - Hwfall - Hmenu - Hstatus - Hmacros - Hqsoframe - Hnotes)
// Htext = HNOM - 140 - Hwfall
#define Hrcvtxt		(Htext) / 2
#define Hxmttxt		(Htext - (Hrcvtxt))

#define Wwfall		(WNOM - Hwfall + BTN_HEIGHT + 2 * BEZEL)
#define Wmode 		80
#define Ws2n  		100
#define Wimd  		100
#define Wwarn 		16
#define bwAfcOnOff	(Hwfall -22)/2
#define bwSqlOnOff	(Hwfall -22)/2

#define Wstatus (WNOM - Wmode - Ws2n - Wimd - bwAfcOnOff - bwSqlOnOff - Wwarn)

//remove the comment delimiter to enable experimental psk250 and qpsk250 modes

#define USE250

//remove the comment delimiter to enable both old and new text widget class
//switching from one to the other requires a restart of the program before
//the newly selected widget becomes effective.
//
//default is to use the new text widget class that is a fully enabled editor
//widget
//#define USE_BOTH_TEXT_WIDGETS


#endif
